pallets/loans/src/lib.rs

```diff
@@ -448,6 +454,28 @@ pub mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        ///Migrate reward data from RewardAccured to RewardAccrued
+        #[pallet::weight(T::WeightInfo::migrate_reward_data())]
+        #[transactional]
+        pub fn migrate_reward_data(_origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            let mut remove_keys = Vec::new();
+            for (account_id, reward_amount) in RewardAccured::<T>::iter() {
+                if reward_amount > 0 {
+                    RewardAccrued::<T>::insert(account_id.clone(), reward_amount);
+                }
+                remove_keys.push(account_id);
+                if remove_keys.len() >= 30 {
+                    break;
+                }
+            }
+
+            for key in remove_keys.iter() {
+                RewardAccured::<T>::remove(key);
+            }
+
+            Ok(().into())
```

Comment:
better to emit some event to tell when the migration finished

Reply:
Bot can check remaining amount before send a `migrate_reward_data` transaction.